### ISSUE TITLE
Docker metrics to Prometheus always report as server: 127.0.0.1:9000 instead of MINIO_SERVER_URL

### ISSUE THREAD

**Issue by odubuc** [minio/minio#19842]
## Expected Behavior
When deploying MinIO using Docker container, one container per host, I expected metrics fetched from prometheus to have the value of "server" to the "MINIO_SERVER_URL" OR a unique configurable name 

## Current Behavior
All 3 containers deployed over three servers have the name "127.0.0.1:9000"
example of metrics fetched using curl on one of the node
```
minio_node_drive_total_bytes{drive="/mnt/data",server="127.0.0.1:9000"}
minio_software_version_info{server="127.0.0.1:9000",version="2024-05-28T17:19:04Z"}
```
When looking at the metrics on MinIO web UI, since every metrics are flaged as 127.0.0.1:9000, it seems it gets confused and reports 1 server. however there are three.

## Steps to Reproduce 
config file used, just changed domain named ( and 1, 2, 3 for MINIO_SERVER_URL on host 1, 2, 3 )
```
MINIO_ROOT_USER=admin
MINIO_ROOT_PASSWORD=change_me

MINIO_VOLUMES="/mnt/data"

MINIO_OPTS="--console-address :9001"

MINIO_SERVER_URL="https://dev-io-1.example.com"
MINIO_BROWSER_REDIRECT_URL="https://minio.dev.example.com/ui"

MINIO_PROMETHEUS_AUTH_TYPE="public"
MINIO_PROMETHEUS_URL="http://prometheus.example.com:9090"
```

launched with 
```
sudo docker run -d \
    --name minio \
    --restart=always \
    -p 9000:9000 -p 9001:9001 \
    -v /mnt/minio/data:/mnt/data:rw \
    -v /opt/minio:/etc/minio:ro \
    -e "MINIO_CONFIG_ENV_FILE=/etc/minio/config.env" \
    minio/minio server;
```

NGINX as reverse proxy configured like specified in the doc 
https://min.io/docs/minio/linux/integrations/setup-nginx-proxy-with-minio.html

Load balancer in from of the 3 hosts at minio.dev.example.com

## Context
Even if I read the doc over 15 times, this is my first deployement of MinIO so maybe I missed a crucial part of the documentation.

## Your Environment

MinIO: 2024-05-28T17:19:04Z
Host: Debian 11
Docker : 26.1.3
uname -a : Linux dev-io-1 5.10.0-29-cloud-amd64 # 1 SMP Debian 5.10.216-1 (2024-05-03) x86_64 GNU/Linux

**Comment by harshavardhana** [comment]
if you want a specific server IP you should provide that via `--address` - this is normal to expect on a single node, single drive setup. 

In distributed setup these point to each distributed nodes.  MINIO_SERVER_URL is not meant for these things.

**Comment by odubuc** [comment]
That would be exactly what I need. Where do you configure --address, I'm having a hard time finding it in the documentation ? 

MINIO_OPTS ?

**Comment by harshavardhana** [comment]
> That would be exactly what I need. Where do you configure --address, I'm having a hard time finding it in the documentation ?
> 
> MINIO_OPTS ?

yeah `MINIO_OPTS` --> minio options passes this value to MinIO as flags

**Comment by odubuc** [comment]
I tried this :
MINIO_OPTS="--console-address :9001 --address :172.16.64.5"
or this :
MINIO_OPTS="--console-address :9001 --address :dev-io-1.example.com"

But the metrics still shows server:127.0.0.1

Between each try I'm destroying the data just in case something persists.

**Comment by ravindk89** [comment]
> I tried this : MINIO_OPTS="--console-address :9001 --address :172.16.64.5" or this : MINIO_OPTS="--console-address :9001 --address :dev-io-1.example.com"
> 
> But the metrics still shows server:127.0.0.1
> 
> Between each try I'm destroying the data just in case something persists.

Not the right syntax - address goes on the left of the `:`. You're setting the port to an IP or hostname, which I actually am surprised didn't throw an error.

**Comment by odubuc** [comment]
oh my, you're correct. Not my greatest moment. I did the following 6 tests based on your comment
 ```
MINIO_OPTS="--console-address :9001 --address 172.16.64.5"
MINIO_OPTS="--console-address :9001 --address 172.16.64.5:"
MINIO_OPTS="--console-address :9001 --address 172.16.64.5:9000"

MINIO_OPTS="--console-address :9001 --address dev-io-1.example.com"
MINIO_OPTS="--console-address :9001 --address dev-io-1.example.com:"
MINIO_OPTS="--console-address :9001 --address dev-io-1.example.com:9000"
```

172.16.64.5 : ip of the host 
dev-io-1.example.com : fqdn of the host

Since I'm using the value of the host inside the minio container, if it tries to bind to that adress I'd understand if it fails however I didn't get any errors. To make sure I test this accordingly, I've changed my deployement to --net=host and tried again the 6 tests however none of them worked. 

Using curl to fetch the metric between each test, I still get 127.0.0.1
```
curl https://dev-io-1.example.com/minio/v2/metrics/node
minio_software_version_info{server="127.0.0.1:9000",version="2024-05-28T17:19:04Z"} 0
```

Am I missing something obvious again ? 

Between each test I'm destroying the container, deleting the data and using curl to validate the output.

**Comment by ravindk89** [comment]
Ah, so I think this might be because the MinIO server itself is listening on localhost, because that's how it starts. Maybe.

Try making sure `MINIO_VOLUMES=http://hostname/mnt/data` for example. You may also want to set `--hostname=hostname` in the docker container.

**Comment by odubuc** [comment]
`--hostname=hostname` didn't change anything unfortunately and setting `MINIO_VOLUMES=http://hostname/mnt/data` on a single node gives me the output 
```
ERROR Invalid command line arguments: use path style endpoint for single node setup
      > Please check the endpoint
      HINT:
        Single-Node modes requires absolute path without hostnames:
      Examples:
         $ minio server /data/minio/ #Single Node Single Drive
         $ minio server /data-{1...4}/minio # Single Node Multi Drive
```
so I don't think I can add the hostname there.

**Comment by ravindk89** [comment]
Apologies - but it does look like `--address` should be working:

```shell
minio server /tmp/minio --address "minio.example.net:9000"                    ✔  5s  
WARNING: Detected default credentials 'minioadmin:minioadmin', we recommend that you change these values with 'MINIO_ROOT_USER' and 'MINIO_ROOT_PASSWORD' environment variables
MinIO Object Storage Server
Copyright: 2015-2023 MinIO, Inc.
License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
Version: RELEASE.2023-12-20T01-00-02Z (go1.21.5 linux/amd64)

Status:         1 Online, 0 Offline. 
S3-API: http://minio.example.net:9000 
RootUser: minioadmin 
RootPass: minioadmin 

Console: http://192.168.1.213:44909 http://172.17.0.1:44909 http://172.18.0.1:44909 http://127.0.0.1:44909          
RootUser: minioadmin 
RootPass: minioadmin 

Command-line: https://min.io/docs/minio/linux/reference/minio-mc.html#quickstart
   $ mc alias set 'myminio' 'http://minio.example.net:9000' 'minioadmin' 'minioadmin'

Documentation: https://min.io/docs/minio/linux/index.html
Warning: The standard parity is set to 0. This can lead to data loss.
```

followed by

```
# HELP minio_node_drive_total Total drives in this node
# TYPE minio_node_drive_total gauge
minio_node_drive_total{server="minio.example.net:9000"} 1
# HELP minio_node_drive_total_bytes Total storage on a drive
# TYPE minio_node_drive_total_bytes gauge
minio_node_drive_total_bytes{drive="/tmp/minio",server="minio.example.net:9000"} 1.6682549248e+10
```

This might be a docker issue of some sort? All I did was set the hostname in `/etc/hosts`, nothing else beyond setting `--address`

**Comment by odubuc** [comment]
Right, since I wanted this to work seemlessly with the host network fqdn + Docker, the working solution I found this morning is : 

1. Set `--net=host` when launching the container.
2. Do NOT use `MINIO_OPTS` inside config.env to set `--address`. Instead provide the arguments directly through the docker args.

Point number 2 is where I struggled the most since I though it was setting `--address` but I was wrong to assume.

In my case, this was not working:
`MINIO_OPTS="--console-address minio.example.net:9001 --address minio.example.net:9000"`

But this is working great :
```
sudo docker run -d \
    --name minio \
    --restart=always \
    --net=host \
    -v /mnt/minio:/mnt/data:rw \
    -v /opt/minio:/etc/minio:ro \
    -e "MINIO_CONFIG_ENV_FILE=/etc/minio/config.env" \
    minio/minio server --console-address minio.example.net:9001 --address minio.example.net:9000;
```

TL;DR: `MINIO_OPTS` doesn't work to set `--address` nor `--console-address` for docker deployment but the workaround is fine.
